src/OrchardCore.Modules/OrchardCore.Taxonomies/Controllers/AdminController.cs

```diff
@@ -218,6 +218,7 @@ public async Task<IActionResult> EditPost(string taxonomyContentItemId, string t
             // Create a new item to take into account the current type definition.
             var contentItem = await _contentManager.NewAsync(existing.ContentType);
 
+            contentItem.ContentItemId = existing.ContentItemId;
```

Comment:
Note: this fix needs to be applied to Menus, Custom Settings, and Custom UserSettings (possibly)